openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/ValidationReportHandler.java

```diff
@@ -119,7 +115,14 @@ private static Optional<String> getNormalizedPath(ValidationReport.Message messa
             .map(apiOperation -> apiOperation.getApiPath().normalised());
     }
 
-    private static Optional<Integer> getResponseStatus(ValidationReport.Message message) {
+    private static Optional<Integer> getResponseStatus(
+        @Nullable ResponseMetaData response,
+        ValidationReport.Message message
+    ) {
+        if (response != null && response.getStatusCode() != null) {
+            return Optional.of(response.getStatusCode());
```

Comment:
Curiosity: if status code is not null why do we return an Optional?

Reply:
Because further below it can still be null -> Optional.empty

Reply:
I need a coffee..thanks @pboos